zio-http/src/main/scala/zio/http/model/headers/values/Allow.scala

```diff
@@ -0,0 +1,35 @@
+package zio.http.model.headers.values
+
+import zio.Chunk
+import zio.http.model.Method
+
+import scala.annotation.tailrec
+
+/**
+ * The Allow header must be sent if the server responds with a 405 Method Not Allowed status code to indicate
+ * which request methods can be used.
+ */
+final case class Allow(methods: Chunk[Method])
```

Comment:
@ccerbusca you have to define the values as ADT. `Method.fromString` is not failing but the return value in case of invalid method is not usable here, as it is not part oh Http standard.

Reply:
In case the list of request methods contains an invalid method, should the element be ignored or the whole list invalidated?

Reply:
no, I should have a `InvalidAllowMethod` value.

`sealed trait Allow
  object Allow {
     case object Get extends Allow
     case object InvalidAllowMethod extends Allow
     .... 
  }
 `

Reply:
Oh, i see. Makes sense

Reply:
I'm not really sure why we can't use `Method`. The RFC does not define the list of accepted methods AFAIK and we allow `CustomMethod`. I think that pretty much You can have anything that matches `[a-zA-Z0-9]` and it should produce correct `Allow` header.

Reply:
E.g. Http4s will allow `DOG, CAT, TEST, 434,` to be correct `Allow` header.

Reply:
That's correct we don't need to follow http4s that was just an example. Even if we don't want to allow anything apart from standard set of HTTP methods, I'm not sure why we want to model them as separate ADT instead of doing `Method.fromString` and failing if we get `CustomMethod` as a result.

Reply:
I refactored the implementation to use an ADT. We could discuss further if you think going back to using Method.fromString is the better solution